airbyte/mcp/cloud_ops.py

```diff
@@ -601,6 +631,111 @@ def get_cloud_sync_status(
     return result
 
 
+@mcp_tool(
+    domain="cloud",
+    read_only=True,
+    idempotent=True,
+    open_world=True,
+    extra_help_text=CLOUD_AUTH_TIP_TEXT,
+)
+def list_cloud_sync_jobs(
+    connection_id: Annotated[
+        str,
+        Field(description="The ID of the Airbyte Cloud connection."),
+    ],
+    *,
+    workspace_id: Annotated[
+        str | None,
+        Field(
+            description=WORKSPACE_ID_TIP_TEXT,
+            default=None,
+        ),
+    ],
+    max_jobs: Annotated[
+        int,
+        Field(
+            description=(
+                "Maximum number of jobs to return. "
+                "Defaults to 20 if not specified. "
+                "If '0' is provided, no limit is applied (uses API default)."
+            ),
+            default=20,
+        ),
+    ],
+    from_tail: Annotated[
+        bool | None,
+        Field(
+            description=(
+                "When True, jobs are ordered newest-first (createdAt DESC). "
+                "When False, jobs are ordered oldest-first (createdAt ASC). "
+                "Defaults to True if `jobs_offset` is not specified. "
+                "Cannot combine `from_tail=True` with `jobs_offset`."
+            ),
+            default=None,
+        ),
+    ],
+    jobs_offset: Annotated[
+        int | None,
+        Field(
+            description=(
+                "Number of jobs to skip from the beginning. "
+                "Cannot be combined with `from_tail=True`."
+            ),
+            default=None,
+        ),
+    ],
+) -> SyncJobListResult:
+    """List sync jobs for a connection with pagination support.
+
+    This tool allows you to retrieve a list of sync jobs for a connection,
+    with control over ordering and pagination. By default, jobs are returned
+    newest-first (from_tail=True).
+    """
+    # Validate that jobs_offset and from_tail are not both set
+    if jobs_offset is not None and from_tail is True:
+        raise PyAirbyteInputError(
+            message="Cannot specify both 'jobs_offset' and 'from_tail=True' parameters.",
+            context={"jobs_offset": jobs_offset, "from_tail": from_tail},
+        )
+
+    # Default to from_tail=True if neither is specified
+    if from_tail is None and jobs_offset is None:
+        from_tail = True
+    elif from_tail is None:
+        from_tail = False
+
+    workspace: CloudWorkspace = _get_cloud_workspace(workspace_id)
+    connection = workspace.get_connection(connection_id=connection_id)
+
+    # Use 0 as "no limit" - pass a high number to the API
+    effective_limit = max_jobs if max_jobs > 0 else 1000
```

Comment:
I think this would overload agent context. Reduce to 500 as "max" behavior, and remove the "0" option for "very high". You can put gt/lt constraints in the annotation description - although I don't know if we need to actually enforce them.